### PR TITLE
Fix pvPortRealloc

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

**Comment by Acuadros95** [comment]
@mergify backport main foxy

**Comment by Acuadros95** [comment]
@jcmonteiro feel free to add comments or request modifications

**Review by pablogs9 — APPROVED** [review]

File: microros_olimex_e407_extensions/Src/custom_memory_manager.c [hunk 1]
```diff
@@ -300,21 +300,27 @@ void *pvPortRealloc( void *pv, size_t xWantedSize )
...
+		char *in_dest = (char*)newmem;
 
-	char *in_src = (char*)pv;
-  	char *in_dest = (char*)newmem;
-	size_t count = pxLink->xBlockSize & ~xBlockAllocatedBit;
+		size_t count = (pxLink->xBlockSize & ~xBlockAllocatedBit) - xHeapStructSize;
+		if (xWantedSize < count)
+		{
+			count = xWantedSize;
+		}
```
Comment (jcmonteiro) [inline microros_olimex_e407_extensions/Src/custom_memory_manager.c:318]: @Acuadros95, maybe it is not my place to comment about formatting (sorry for nitpicking :grimacing:), but would it make sense to convert this `if` statement to a one-liner like the `while` statement below? Or the other way around, add the brackets to the `while` statement.

**Review by jcmonteiro — APPROVED** [review]

File: microros_olimex_e407_extensions/Src/custom_memory_manager.c (hunk 1 above)
Comment (Acuadros95) [reply microros_olimex_e407_extensions/Src/custom_memory_manager.c]: No problem: 7f9eb5371ad284f47a004e500cea5c196598e2ce

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**